test/saithrift/tests/saiacl.py

```diff
@@ -1634,3 +1634,453 @@ def runTest(self):
+          14. Create two ACL table to block the traffic based on Src IP address (192.168.0.1) and Dst IP address (12.12.12.1).
+          15. Create two ACL table to block the traffic based on Src IP address (10.10.10.1) and Dst IP address (11.11.11.1).
+          16. Associate all the ACL tables with the group members.
+          17. Bind the ACL table with VLAN attribute SAI_VLAN_ATTR_INGRESS_ACL and LAG attribute SAI_LAG_ATTR_INGRESS_ACL
+          18. Send the IP packet again through port 1 and 2 and ensure the packets are dropped at Ingress of LAG and VLAN.
+
+        Part3:
+          19. Remove all the ACL table and the entries bound with VLAN and LAG port
+          20. Send the packet from port 1 and 2 and ensure the traffic forwarded for VLAN and LAG.
+          21. Remove the route and the associated router interface, vlan members and the corresponding VLAN from the database.
+        """
+
+        switch_init(self.client)
+        port1 = port_list[0]
+        port2 = port_list[1]
+        v4_enabled = 1
+        v6_enabled = 0
+        vlan_id = 100
+        mac_action = SAI_PACKET_ACTION_FORWARD
+        mac1 = ''
+
+        # Creating VLAN
+        vlan_oid = sai_thrift_create_vlan(self.client, vlan_id)
+        vlan_member1 = sai_thrift_create_vlan_member(self.client, vlan_oid, port2, SAI_VLAN_TAGGING_MODE_UNTAGGED)
+
+        attr_value = sai_thrift_attribute_value_t(u16=vlan_id)
+        attr = sai_thrift_attribute_t(id=SAI_PORT_ATTR_PORT_VLAN_ID, value=attr_value)
+        self.client.sai_thrift_set_port_attribute(port2, attr)
+
+        # Creating VR and LAG
+        vr_id = sai_thrift_create_virtual_router(self.client, v4_enabled, v6_enabled)
+
+        lag_id1 = self.client.sai_thrift_create_lag([])
+        sai_thrift_vlan_remove_ports(self.client, switch.default_vlan.oid, [port1])
+        lag_member_id1 = sai_thrift_create_lag_member(self.client, lag_id1, port1)
+
+        # Creating RIF
+        rif_id1 = sai_thrift_create_router_interface(self.client, vr_id, SAI_ROUTER_INTERFACE_TYPE_PORT, lag_id1, 0, v4_enabled, v6_enabled, mac1)
+        rif_id2 = sai_thrift_create_router_interface(self.client, vr_id, SAI_ROUTER_INTERFACE_TYPE_VLAN, 0, vlan_oid, v4_enabled, v6_enabled, mac1)
+
+        addr_family = SAI_IP_ADDR_FAMILY_IPV4
+        ip_addr1 = '10.10.10.1'
+        ip_addr_subnet1 = '11.11.11.0'
+        ip_mask = '255.255.255.0'
+        dmac1 = '00:11:22:33:44:55'
+        ip_addr2 = '192.168.0.1'
+        ip_addr_subnet2 = '12.12.12.0'
+        dmac2 = '00:11:22:33:44:56'
+
+        # Creating Neighbor, NHop and Route
+        sai_thrift_create_neighbor(self.client, addr_family, rif_id2, ip_addr2, dmac2)
+        sai_thrift_create_neighbor(self.client, addr_family, rif_id1, ip_addr1, dmac1)
+
+        nhop1 = sai_thrift_create_nhop(self.client, addr_family, ip_addr2, rif_id2)
+        nhop2 = sai_thrift_create_nhop(self.client, addr_family, ip_addr1, rif_id1)
+
```

Comment:
Test is passing after adding local route 10.10.10.0/24 to rif_id1 and local route 192.168.0.0/16 to rif_id2

Reply:
Addressed review comments